data/tutorial/dune

```diff
@@ -0,0 +1,11 @@
+(rule
+  (targets mdx.ml)
+  (deps (:markdown (glob_files *.md)))
+  (action (with-stdout-to %{targets}
+            (bash "echo %{markdown} | xargs --max-args=1 ocaml-mdx pp")))
```

Comment:
If you use `ocaml-mdx test` it should show you diffs and you can use dune promotion rules. Check with @clecat to find the correct runes.

Reply:
Hello there, I gave a quick look at this and we have some problems:
The markdown files cuts a lot the modules they defines, puting commentaries in-between.
We end up with mdx yelling about incorrect syntax (as we have `struct` tags without `end`)
I think we should at a way to support that behaviour (like telling mdx that the current block is attached to the latest) before using ocaml-mdx there.
Craig will open an issue about it, I let him put the link here

Reply:
Note that `ocaml-mdx pp` might also be the right thing to do here, as we are trying to compile the examples into a working executable.

Reply:
There are limitations with `ocaml-mdx test` which prevent it from being usable in this case (yet!). I've opened https://github.com/realworldocaml/mdx/issues/158 to track this.

It's also not clear to me why we'd want a diff/promotion workflow here; is this better than simply running the output of `pp`?

Reply:
Oops, my reply happened concurrently with yours; sorry! I wish Github would warn me about this.